File: backend/app/schemas/ticket.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field, field_serializer


class TicketCreate(BaseModel):
    title: str = Field(..., min_length=3, max_length=200)
    description: str = Field(..., min_length=10, max_length=5000)


class TicketResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    description: str
    category: str | None = None
    priority: str | None = None
    summary: str | None = None
    ai_confidence: float | None = None
    created_at: datetime
    updated_at: datetime

    @field_serializer("ai_confidence")
    def serialize_confidence(self, value: Decimal | float | None) -> float | None:
        if value is None:
            return None
        return float(value)


class TicketListResponse(BaseModel):
    items: list[TicketResponse]
    total: int
    page: int
    page_size: int


class CategoryResponse(BaseModel):
    name: str

File: frontend/src/types/ticket.ts
export interface Ticket {
  id: string;
  title: string;
  description: string;
  category?: string | null;
  priority?: string | null;
  summary?: string | null;
  ai_confidence?: number | null;
  created_at: string;
  updated_at: string;
}

export interface TicketAnalyzeResponse extends Ticket {
  inference_source?: string | null;
  confidence_breakdown?: Record<string, number> | null;
  processing_ms?: number | null;
}

export interface TicketCreate {
  title: string;
  description: string;
}

export interface TicketListResponse {
  items: Ticket[];
  total: number;
  page: number;
  page_size: number;
}

export interface DashboardStats {
  total_tickets: number;
  analyzed_total: number;
  pending_total: number;
  analyzed_today: number;
  analysis_rate: number;
  by_category: Record<string, number>;
  by_priority: Record<string, number>;
  avg_confidence?: number | null;
}

export interface TicketFilters {
  page?: number;
  page_size?: number;
  analyzed?: string;
  category?: string;
  priority?: string;
  search?: string;
}
